docs/docs/code/python/auth/README.md

```diff
@@ -0,0 +1,103 @@
+thumbnail: https://res.cloudinary.com/pipedreamin/image/upload/v1646763806/docs/icons/icons8-connected-96_fcbhxc.png
+---
+
+# Connecting apps in Python
+
+When you use [prebuilt actions](/components#actions) tied to apps, you don't need to write the code to authorize API requests. Just [connect your account](/connected-accounts/#connecting-accounts) for that app and run your workflow.
+
+But sometimes you'll need to [write your own code](/code/python/). You can also connect apps to custom code steps, using the auth information to authorize requests to that app.
+
+For example, you may want to send a Slack message from a step. We use Slack's OAuth integration to authorize sending messages from your workflows. 
+
+Add Slack as an app on the Python step, then connect your Slack account.
+
+![Add your Slack account to a Python code step by adding it](https://res.cloudinary.com/pipedreamin/image/upload/v1658954165/docs/components/CleanShot_2022-07-27_at_16.35.37_ytofp2.gif)
+
+Then within the Python code step, `pd.steps["slack"]["$auth"]["oauth_access_token"]` will contain your Slack account OAuth token.
+
+With that token, you can make authenticated API calls to Slack:
+
+```python
+from slack_sdk import WebClient
+
+def handler(pd: "pipedream"):
+  # Your Slack OAuth token is available under pd.inputs
+  token = pd.inputs["slack"]["$auth"]["oauth_access_token"]
+  
+  # Instantiate a new Slack client with your token
+  client = WebClient(token=token)
+
+  # Use the client to send messages to Slack channels
+  response = client.chat_postMessage(
+    channel='#pierces-playground',
+    text='Hello from Pipedream!'
+  )
+  
+  # Export the Slack response payload for use in future steps
+  pd.export("response", response.data)
+```
+
+
+[[toc]]
+
+## Accessing connected account data with `pd.inputs[appName]["$auth"]`
+
+In our Slack example above, we created a Slack `WebClient` using the Slack OAuth access token:
+
+```python
+  # Instantiate a new Slack client with your token
+  client = WebClient(token=token)
+```
+
+Where did `pd.inputs["slack"]` come from? Good question. It was generated when we connected Slack to our Python step.
+
+![The Slack app generates the pd.inputs["slack"] data](https://res.cloudinary.com/pipedreamin/image/upload/v1658954351/docs/components/CleanShot_2022-07-27_at_16.38.42_f08ocy.png)
+
+The Slack access token is generated by Pipedream, and is available to this step in the `pd.inputs[appName]["$auth"]` object:
+
+```python
+from slack_sdk import WebClient
+
+def handler(pd: "pipedream"):
+  token = pd.inputs["slack"]["$auth"]["oauth_access_token"]
+  # Authentication details for all of your apps are accessible under the special pd.inputs["slack"] variable:
+  console.log(pd.inputs["slack"]["$auth"])
+```
+
+`this.appName.$auth` contains named properties for each account you connect to the associated step. Here, we connected Slack, so `this.slack.$auth` contains the Slack auth info (the `oauth_access_token`).
```

Comment:
References `this.app.$auth` which isn't exposed in Python right?

Reply:
Nice catch!

Reply:
Found some other invalid node references, fixing those.